lib/jsonapi/authorization/authorizing_processor.rb

```diff
@@ -181,28 +182,35 @@ def authorize_replace_to_many_relationship
         )
       end
 
-      def authorize_remove_to_many_relationship
+      def authorize_remove_to_many_relationships
         source_resource = @resource_klass.find_by_key(
           params[:resource_id],
           context: context
         )
         source_record = source_resource._model
 
-        relationship_type = params[:relationship_type].to_sym
-        related_resource = @resource_klass
-          ._relationship(relationship_type)
+        related_resources = @resource_klass
+          ._relationship(params[:relationship_type].to_sym)
           .resource_klass
-          .find_by_key(
-            params[:associated_key],
+          .find_by_keys(
+            params[:associated_keys],
             context: context
           )
-        related_record = related_resource._model unless related_resource.nil?
 
-        authorizer.remove_to_many_relationship(
-          source_record,
-          related_record,
-          relationship_type
-        )
+        # find_by_keys doesn't raise if no records, so raise explicitly here
+        if related_resources.empty?
+          fail JSONAPI::Exceptions::RecordNotFound.new(params[:associated_keys])
+        else
+          related_records = related_resources.map { |resource| resource._model }
+        end
+
+        related_records.each do |related_record|
+          authorizer.remove_to_many_relationship(
+            source_record,
+            related_record,
+            params[:relationship_type]
+          )
```

Comment:
Instead of looping through all the `related_records`, we might want to instead change the API of `remove_to_many_relationship` to take an array of `related_records` (instead of one at a time).

That way we wouldn't even have to do the `if related_resource.empty?` check you're doing before calling the authorizer right now, too.

There was a comment https://github.com/venuu/jsonapi-authorization/pull/40#issuecomment-280493499 that maybe related to this not being possible before 0.9 version of JR.